.github/workflows/publish-docs.yaml

```diff
@@ -35,6 +46,38 @@ jobs:
           path: docs
           ref: ${{ env.DOCS_BRANCH }}
           token: ${{ secrets.GH_RAD_CI_BOT_PAT }}
+      # Setup dependencies
+      - name: Set up Go ${{ env.GOVER }}
+        uses: actions/setup-go@v4
+        with:
+          go-version: ${{ env.GOVER }}
+          cache-dependency-path: radius/go.sum
+      - name: Setup NodeJS
+        uses: actions/setup-node@v3
+        with:
+          node-version: '18'
+      - name: Install TypeSpec compiler
+        run: |
+          pushd radius/typespec
+          npm ci
+          popd
+      - name: Install autorest
+        run: npm install -g autorest@3.6.1
+      - name: Install oav
+        run: npm install -g oav@3.2.0
+      - name: Install mockgen
+        run: |
+          cd radius
+          go install github.com/golang/mock/mockgen@v1.5.0
+      - name: Install controller-gen
+        run: |
+          cd radius
+          go install sigs.k8s.io/controller-tools/cmd/controller-gen@v0.9.1
```

Comment:
nonblocking for your PR: we should really simplify this. These steps have been copied to a few places now.